blocklib/timing/src/event_definitions.hpp

```diff
@@ -0,0 +1,428 @@
+        {352, {"CMD_UNI_BREQ", "Request UNILAC PZ to deliver beam"}},
+        {353, {"CMD_UNI_BPREP", "Request UNILAC PZ to prepare beam delivery"}},
+        {354, {"CMD_UNI_BREQ_NOWAIT", "Request UNILAC PZ to deliver beam but without 'wait block' in Data Master schedule"}},
+        {512, {"CMD_FG_PREP", "Prepare function generator"}},
+        {513, {"CMD_FG_START", "Start function generator"}},
+        {518, {"CMD_BEAM_ON", "General: Begin of beam passage"}},
+        {520, {"CMD_BEAM_OFF", "General: End of beam passage"}},
+        {521, {"CMD_SEPTUM_CHARGE", "Start septum ramp up"}},
+        {522, {"CMD_EBEAM_ON", "Electron beam on"}},
+        {523, {"CMD_EBEAM_OFF", "Electron beam off"}},
+        {524, {"CMD_STOCH_COOLING_ON", "Stochastic cooling on"}},
+        {525, {"CMD_STOCH_COOLING_OFF", "Stochastic cooling off"}},
+        {526, {"CMD_TARGET_ON", "Enable target"}},
+        {527, {"CMD_TARGET_OFF", "Disable target"}},
+        {528, {"CMD_EXP_DAQ_START", "Start experiment data acquisition"}},
+        {529, {"CMD_EXP_DAQ_STOP", "Stop experiment data acquisition"}},
+        {530, {"CMD_EXP_DRIVE_IN", "Move experiment actuators (detectors) to measurement position"}},
+        {531, {"CMD_EXP_DRIVE_OUT", "Move experiment actuators (detectors) to outside position (clear beam aperture)"}},
+        {532, {"CMD_DRIVE_MOVE", "Move timing-controlled actuators to new position"}},
+        {539, {"CMD_CHOPPER_CHARGE", "Start energizing chopper"}},
+        {1024, {"CMD_BUMPER_CHARGE", "Start bumper magnet ramp up"}},
+        {1025, {"CMD_BUMPER_START", "Start bumper ramp down"}},
+        {1026, {"CMD_INJ_KICKER_START", "Start injection kicker"}},
+        {1027, {"CMD_EXTR_KICKER_START", "Start extraction kicker"}},
+        {2048, {"CMD_B2B_PMEXT", "B2B internal: request phase measurement (extraction)"}},
+        {2049, {"CMD_B2B_PMINJ", "B2B internal: request phase measurement (injection)"}},
+        {2050, {"CMD_B2B_PREXT", "B2B internal: send result of phase measurement (extraction)"}},
+        {2051, {"CMD_B2B_PRINJ", "B2B internal: send result of phase measurement (injection)"}},
+        {2052, {"CMD_B2B_TRIGGEREXT", "B2B: trigger kicker electronics (extraction)"}},
+        {2053, {"CMD_B2B_TRIGGERINJ", "B2B: trigger kicker electronics (injection)"}},
+        {2054, {"CMD_B2B_DIAGKICKEXT", "B2B: send result of kick diagnostic (extraction)"}},
+        {2055, {"CMD_B2B_DIAGKICKINJ", "B2B: send result of kick diagnostic (injection)"}},
+        {2056, {"CMD_B2B_DIAGEXT", "B2B internal: send result of optional diagnostic (extraction)"}},
+        {2057, {"CMD_B2B_DIAGINJ", "B2B internal: send result of optional diagnostic (injection)"}},
+        {2079, {"CMD_B2B_START", "Start B2B procedure"}},
+        {4000, {"CMD_GMT_INTERNAL1", "internal use by the GMT"}}
+};
+
+static const std::map<std::string, std::string> demoSchedules = {{"Pattern 1", R"(
```

Comment:
This simulates a Timing schedule via event injection, right ?

You might want to make it a more readable instead of using hex codes ... I currently use a bash script for timing simulation, though it probably helps to get an idea:

https://gitlab.com/al.schwinn/DigitizerDU2/-/blob/master/scripts/inject_wr_timing/Inject_simulation.sh?ref_type=heads#L35

The 'inject' method builds the hex-code from the passed  GroupID, EventID, SequenceID, ProcessID, ChainID, Flags, according to the specified standard and injects the event via `saft-ctl`

Edit: looks like the class Timing::event already provides such a service. Maybe should be a static method ?

Reply:
Yes. The format is the same as the one used by the `saft-dm` tool and I don't suppose anyone will edit this by hand. The UI tool can export this format to the clipboard so it's possible to edit it there and just paste it there.